### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by johnkioko**
**In this project, I have set up tests on the controllers I recently added to your Blog app.**

Added RSpec as a gem to your project.
Created a Request spec file for all controllers.
Made sure that for each action, I checked:

- If response status was correct.

- If a correct template was rendered.

- If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      get '/users/1'
+      expect(response).to render_template('show')
+    end
+  end
+  context 'Response Content' do
+    it 'includes "here are all users" for root endpoint' do
+      get '/'
+      expect(response.body).to include('here are all users')
+    end
+    it 'includes "here are individual users" for user endpoint' do
```
Comment (omar-muhamad) [inline spec/requests/users_spec.rb]: - [ ] Please include tests for `users index` and `users show` separately inside `describe` method like that:

```ruby
describe 'GET /index' do
  # users index tests here
end

describe 'GET /show' do
  # users show tests here
end
```

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  describe 'GET /index' do
+    before { get '/users/1/posts' }
+
+    it 'has a 200 OK status' do
+      expect(response).to have_http_status(200)
+    end
+    it "renders the 'index' view" do
+      expect(response).to render_template('index')
+    end
+  end
```
Comment (omar-muhamad) [inline spec/requests/posts_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20the%20response%20body%20includes%20correct%20placeholder%20text.), Please add a test for `posts index` asserts that the response body includes correct placeholder text.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @johnkioko,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please fill out all sections of `getting started` section in the image below with the proper data and commands instead of leaving it empty to make your readme file more professional.

image

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @johnkioko ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**